pkg/updater/service/apm_injector.go

```diff
@@ -0,0 +1,178 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !windows
+
+// Package service provides a way to interact with os services
+package service
+
+import (
+	"os"
+	"path"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// SetupAPMInjector sets up the injector at bootstrap
+func SetupAPMInjector() error {
+	injectorPath := "/opt/datadog-packages/datadog-apm-inject/stable"
+	err := os.Chmod(path.Join(injectorPath, "run"), 0777)
+	if err != nil {
+		return err
+	}
+
+	err = setupLDPreload(injectorPath)
+	if err != nil {
+		return err
+	}
+
+	err = setupDockerDaemon(injectorPath)
+	if err != nil {
+		return err
```

Comment:
on each err, shouldn't you cleanup especially LDPreload?

Reply:
Done in [fix(installer): Cleanup APM injector on setup failure](https://github.com/DataDog/datadog-agent/pull/24372/commits/ecba82511581b82fda25ef85e7aa8c4940d76b43). There may just be a side effect that if the `ld.so.preload` write fails, the uninstall may fail for the same reason (e.g. permission issues) and that will emit a log but nothing should be written, hence the _warn_